decentraland/unity-renderer
# chore: add simple test for addressables duplicates

## What does this PR change?
Closes #4366

- Add validation tests, which are based on Addressables Analyze API (for more info on this decision read comments in the #4366).
- Tests are separated for Custom bundles, Resources and Scenes (same as in Addressables Analyze, re-use same Rules)
- Step in CI to run validation only for Custom bundles. Because Resources and Scenes bundles have too many duplicates. We will need to fix them first before re-enabling these tests
- Made automated fix for Duplicates in our bundles (but we will need to refactor this big group for better performance in future)
- Marked several failing Edit Mode tests as "Legacy" (Category), so we will be able to exclude them from the EditMode test Suite in future.
- Add EditModeCI Category to unlock possibility for EditMode tests in CI

## How to test the changes?

[Devs Only]
0. Open Editor -> TestRunner

1. Run AddressablesValidationTests
2. Validate that test for Custom Bundles is passing, but one for Resources and Scenes are failing.

1. Select only EditMode tests and choose Uncategorised in Category 
2. Run Tests and Validate that all 157 tests pass 

- Check in the CI artifacts that `editmode-results.xml` contains `ValidateDuplicateBundleDependencies` with "passed" status, similar to this:
![Uploading image.png…]()


## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
